### PR TITLE
category crud

### PR DESCRIPTION
**Pull request by mody100000**

### CODE REVIEW

File: src/controllers/productController.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
-const errorHandler = require("../utils/errorHandler")
-const validateCreateProductInput = require("../validation/createProduct")
-const productService = require('../services/productService');
+const errorHandler = require('../utils/errorHandler')
+const validateCreateProductInput = require('../validation/createProduct')
+const productService = require('../services/productService')
 
-const createProduct = async (req , res) => {
-    const {success , error , value} = validateCreateProductInput(req.body).validate()
```
Comment (mohammedahmed18) [inline src/controllers/productController.js:6]: we can later modify this validate function to return only error , value for simplicity
then we can check if there is a validation error if the error object is defined

File: src/models/Category.js [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
+    required: true,
+  },
+  icon: {
+    type: String,
+    default: DEFAULT_CATEGORY_ICON,
+  },
+  products: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Product' }],
+})
 
-} , {timestamps : true})
```
Comment (mohammedahmed18) [inline src/models/Category.js:17]: > I removed this timestamps by mistake 😂 😂  @mody100000

File: src/routes/productRoutes.js [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+const {
+  getAllProducts,
+  getProduct,
+  deleteProduct,
+} = require('../services/productService')
+
+const router = Router()
+
+router.post('/', productController.createProduct)
+router.get('/', getAllProducts)
```
Comment (mohammedahmed18) [inline src/routes/productRoutes.js:12]: as we said in the next pull request you should refactor the code to include the business logic in the productService not in productController

**Review by mohammedahmed18 — APPROVED** [review]
let's merge this for now and fix the issues I addressed in the next pull request

File: src/controllers/productController.js (hunk 1 above)
Comment (mody100000) [reply src/controllers/productController.js:6]: do u mean to put it in try-catch block and catch an error if there exist ? xD

**Pull request merged**